src/units/unit.cpp

```diff
@@ -1745,54 +1745,46 @@ std::string unit::describe_builtin_effect(std::string apply_to, const config& ef
 		}
 	} else if(apply_to == "hitpoints") {
 		const std::string &increase_total = effect["increase_total"];
-
 		if(!increase_total.empty()) {
-			return utils::print_modifier(increase_total) + " " +
-			t_string(N_("HP"), "wesnoth");
-		}
-	} else if(apply_to == "movement") {
-		const std::string &increase = effect["increase"];
-
-		if(!increase.empty()) {
-			int n = std::stoi(increase);
-			return utils::print_modifier(increase) + " " +
-				_n("move", "moves", n);
-		}
-	} else if(apply_to == "vision") {
-		const std::string &increase = effect["increase"];
-
-		if(!increase.empty()) {
-			return utils::print_modifier(increase) + " " + t_string(N_("vision"), "wesnoth");
+			return vgettext(
+				"wesnoth",
+				"$number_or_percent HP",
+				utils::string_map({{"number_or_percent", utils::print_modifier(increase_total)}}));
 		}
-	} else if(apply_to == "jamming") {
-		const std::string &increase = effect["increase"];
-
-		if(!increase.empty()) {
-			return utils::print_modifier(increase) + " " + t_string(N_("jamming"), "wesnoth");
-		}
-	} else if(apply_to == "max_experience") {
-		const std::string &increase = effect["increase"];
-
-		if(!increase.empty()) {
-			return utils::print_modifier(increase) + " " +
-			t_string(N_("XP to advance"), "wesnoth");
-		}
-	} else if (apply_to == "max_attacks") {
+	} else {
```

Comment:
We can gain an indentation level here if we lose the "else" - IMO that would make the code structure a bit weird though.